crates/amaru-ledger/src/rules.rs

```diff
@@ -0,0 +1,193 @@
+mod block;
+
+use amaru_kernel::{alonzo::MaybeIndefArray, cbor, Block, Hash, Hasher, MintedBlock, Redeemers};
+
+use block::{
+    body_size::{block_body_size_valid, BlockBodySizeTooBig},
+    ex_units::*,
+    header_size::{block_header_size_valid, BlockHeaderSizeTooBig},
+};
+
+use crate::protocol_parameters::ProtocolParameters;
+
+pub enum RuleViolation {
+    BlockBodySizeTooBig(BlockBodySizeTooBig),
+    BlockHeaderSizeTooBig(BlockHeaderSizeTooBig),
+    TooManyExUnitsBlock(TooManyExUnits),
+}
+
+pub enum BlockValidationError {
```

Comment:
We use `thiserror` for errors. Here you probably would want to have RuleViolation a `kind`.
This would also probably remove the need for a bunch of `map_err` and `From` implementations.

Reply:
We're going to improve error handling in a future PR, so this is out of scope.